src/main/kotlin/com/jetbrains/snakecharm/inspections/SnakemakeRuleKeywordsAfterExecutionInspection.kt

```diff
@@ -0,0 +1,45 @@
+package com.jetbrains.snakecharm.inspections
+
+import com.intellij.codeInspection.LocalInspectionToolSession
+import com.intellij.codeInspection.ProblemsHolder
+import com.jetbrains.snakecharm.SnakemakeBundle
+import com.jetbrains.snakecharm.lang.SnakemakeLanguageDialect
+import com.jetbrains.snakecharm.lang.psi.SMKRule
+import com.jetbrains.snakecharm.lang.psi.SMKRuleParameterListStatement
+
+class SnakemakeRuleKeywordsAfterExecutionInspection : SnakemakeInspection() {
+    override fun buildVisitor(
+            holder: ProblemsHolder,
+            isOnTheFly: Boolean,
+            session: LocalInspectionToolSession
+    ) = object : SnakemakeInspectionVisitor(holder, session) {
+        override fun visitSMKRule(smkRule: SMKRule) {
+            if (smkRule.containingFile.language != SnakemakeLanguageDialect) {
+                return
+            }
+
+            var executionSectionOccured = false
+            var executionSectionName = "execution"
```

Comment:
Why do we need such initialization? The value will be always overridden with the correct one.